PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -463,3 +477,70 @@ function isNullOrWhitespace(input: any) {
 function isValidPort(port : any) {
     return Number.isSafeInteger(port) && (port >= 0 && port <= 65535);
 }
+
+function getRulesDocs(): Promise<void> {
+    if (ruleAnalyzerPairs.size > 0 && isDtpSettingsValid) {
+        return doGetRulesDocs(0)
+            .catch((error) => {
+                rulesDocs.clear();
+                if (error === 401) {
+                    tl.warning("You are not authorized to use DTP API.");
+                } else {
+                    tl.warning("Failed to connect to DTP server.");
+                }
+            });
+    }
+    return Promise.resolve();
+}
+
+function doGetRulesDocs(index: number): Promise<any> {
+    const ruleId = Array.from(ruleAnalyzerPairs.keys())[index];
+    // AnalyzerId cannot be undefined.
+    const analyzerId = ruleAnalyzerPairs.get(ruleId) || "";
+
+    return getRuleDoc(ruleId, analyzerId, 1.6)
+        .then((response) => {
+            rulesDocs.set(ruleId, response.data.docsUrl);
+            if(index != ruleAnalyzerPairs.size - 1) {
+                return doGetRulesDocs(++index);
+            } else {
+                return Promise.resolve();
+            }
+        }).catch((error) => {
+            let status = error.response ? error.response.status : 500;
+            if (status === 404) {
+                return getRuleDoc(ruleId, analyzerId, 1)
+                    .then((result) => {
+                        rulesDocs.set(ruleId, result.data.docsUrl);
+                        if(index != ruleAnalyzerPairs.size - 1) {
+                            return doGetRulesDocs(++index);
+                        } else {
+                            return Promise.resolve();
+                        }
+                    }).catch((e) => {
+                        status = e.response ? e.response.status : 500;
+                        if(status === 404) {
+                            rulesDocs.set(ruleId, "");
+                            return doGetRulesDocs(++index);
+                        } else {
+                            return Promise.reject(status);
+                        }
+                    });
+            } else {
+                return Promise.reject(status);
+            }
+        });
+}
+
+function getRuleDoc(ruleId: string, analyzerId: string, apiVersion: number): Promise<any> {
+    const credentials = Buffer.from(`${dtpUsername}:${dtpPassword}`).toString('base64');
+    let url = dtpBaseUrl+ "grs/api/v" + apiVersion +"/rules/doc?rule=" + ruleId + "&analyzerId=" + analyzerId;
+    return axios.default.get(url, {
+        headers: {
+            'Authorization': `Basic ${credentials}`
+        },
+        httpsAgent: new https.Agent({
+            rejectUnauthorized: false
```

Comment:
Disabling the SSL/TLS certificate verification may not be good. We can create separate task to figure out how to handle this better.